pkg/operator/controllers/dnsmasq/machineconfig_controller.go

```diff
@@ -57,6 +58,15 @@ func (r *MachineConfigReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
```

Comment:
Any reason not to put these checks in the very beginning of the `Reconcile`?

Reply:
I don't remember why I chose not to.  

I'm going to put it at the top because that makes the most sense.